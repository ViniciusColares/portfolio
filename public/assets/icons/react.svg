<svg width="15" height="15" viewBox="0 0 15 15" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M7.5 6.16313C7.32444 6.16313 7.1506 6.19771 6.9884 6.26489C6.8262 6.33208 6.67883 6.43055 6.55469 6.55469C6.43055 6.67883 6.33207 6.82621 6.26489 6.9884C6.1977 7.1506 6.16313 7.32444 6.16313 7.5C6.16313 7.67556 6.1977 7.84941 6.26489 8.0116C6.33207 8.1738 6.43055 8.32118 6.55469 8.44532C6.67883 8.56946 6.8262 8.66793 6.9884 8.73511C7.1506 8.8023 7.32444 8.83688 7.5 8.83688C7.67556 8.83688 7.8494 8.8023 8.0116 8.73511C8.1738 8.66793 8.32117 8.56946 8.44531 8.44532C8.56945 8.32118 8.66793 8.1738 8.73511 8.0116C8.8023 7.84941 8.83687 7.67556 8.83687 7.5C8.83687 7.32444 8.8023 7.1506 8.73511 6.9884C8.66793 6.82621 8.56945 6.67883 8.44531 6.55469C8.32117 6.43055 8.1738 6.33208 8.0116 6.26489C7.8494 6.19771 7.67556 6.16313 7.5 6.16313ZM3.755 10.1594L3.46 10.0844C1.26125 9.52875 0 8.58563 0 7.4975C0 6.40938 1.26125 5.46625 3.46 4.91063L3.755 4.83625L3.83812 5.12875C4.06086 5.89607 4.34577 6.64397 4.69 7.365L4.75313 7.49813L4.69 7.63125C4.34519 8.35204 4.06027 9.09998 3.83812 9.8675L3.755 10.1594ZM3.32312 5.59375C1.65187 6.06313 0.62625 6.78125 0.62625 7.4975C0.62625 8.21313 1.65187 8.93125 3.32312 9.40125C3.52752 8.75121 3.77429 8.11526 4.06187 7.4975C3.77397 6.87988 3.52718 6.24391 3.32312 5.59375ZM11.245 10.1594L11.1619 9.86625C10.9394 9.09897 10.6542 8.35125 10.3094 7.63063L10.2462 7.4975L10.3094 7.36438C10.6541 6.64348 10.9392 5.89556 11.1619 5.12813L11.245 4.83563L11.5406 4.91C13.7388 5.46563 15 6.40875 15 7.4975C15 8.58625 13.7388 9.52875 11.5406 10.0844L11.245 10.1594ZM10.9381 7.4975C11.2381 8.14688 11.4862 8.785 11.6769 9.40125C13.3488 8.93125 14.3737 8.21313 14.3737 7.4975C14.3737 6.78125 13.3481 6.06375 11.6769 5.59375C11.4728 6.24389 11.226 6.87986 10.9381 7.4975ZM3.31875 5.59063L3.23563 5.29875C2.6175 3.12 2.805 1.55875 3.75 1.01375C4.67688 0.478752 6.165 1.11063 7.72438 2.71125L7.93688 2.92938L7.72438 3.1475C7.17029 3.72351 6.66403 4.34368 6.21063 5.00188L6.12625 5.1225L5.97938 5.135C5.18182 5.19788 4.39068 5.32538 3.61375 5.51625L3.31875 5.59063ZM4.50375 1.44688C4.33625 1.44688 4.18813 1.48313 4.06313 1.555C3.44187 1.91313 3.33187 3.15813 3.76 4.83813C4.42602 4.68996 5.101 4.58536 5.78062 4.525C6.17276 3.96743 6.60119 3.43628 7.06312 2.935C6.08812 1.98563 5.165 1.44688 4.50375 1.44688ZM10.4969 14.1731C10.4963 14.1731 10.4963 14.1731 10.4969 14.1731C9.60625 14.1731 8.4625 13.5025 7.27562 12.2838L7.06312 12.0656L7.27562 11.8475C7.82957 11.2713 8.33562 10.6509 8.78875 9.9925L8.87313 9.87188L9.01937 9.85938C9.81734 9.79685 10.6089 9.66956 11.3863 9.47875L11.6813 9.40438L11.765 9.69688C12.3819 11.8744 12.195 13.4363 11.25 13.9806C11.0205 14.1104 10.7605 14.1768 10.4969 14.1731ZM7.93688 12.06C8.91187 13.0094 9.835 13.5481 10.4963 13.5481H10.4969C10.6637 13.5481 10.8125 13.5119 10.9369 13.44C11.5581 13.0819 11.6687 11.8363 11.24 10.1563C10.5738 10.3046 9.89861 10.4092 9.21875 10.4694C8.82709 11.0273 8.39885 11.5587 7.93688 12.06ZM11.6813 5.59063L11.3863 5.51625C10.609 5.32483 9.81741 5.19732 9.01937 5.135L8.87313 5.1225L8.78875 5.00188C8.33585 4.34347 7.82979 3.72328 7.27562 3.1475L7.06312 2.92938L7.27562 2.71125C8.83437 1.11125 10.3219 0.479377 11.25 1.01375C12.195 1.55875 12.3825 3.12 11.765 5.29813L11.6813 5.59063ZM9.21875 4.525C9.9325 4.59 10.6106 4.69563 11.24 4.83813C11.6687 3.15813 11.5581 1.91313 10.9369 1.555C10.3194 1.19813 9.15875 1.745 7.93688 2.935C8.39859 3.4363 8.82681 3.96745 9.21875 4.525ZM4.50375 14.1731C4.23987 14.177 3.97973 14.1105 3.75 13.9806C2.805 13.4363 2.6175 11.875 3.23563 9.69688L3.31813 9.40438L3.61313 9.47875C4.335 9.66063 5.13125 9.78875 5.97875 9.85938L6.12562 9.87188L6.20938 9.9925C6.66285 10.6508 7.1691 11.2712 7.72313 11.8475L7.93562 12.0656L7.72313 12.2838C6.53688 13.5025 5.39313 14.1731 4.50375 14.1731ZM3.76 10.1563C3.33125 11.8363 3.44187 13.0819 4.06313 13.44C4.68 13.7919 5.84 13.2494 7.06312 12.06C6.60138 11.5583 6.17296 11.027 5.78062 10.4694C5.10096 10.4093 4.42597 10.3047 3.76 10.1563ZM7.5 10.5488C6.98562 10.5488 6.45688 10.5263 5.9275 10.4825L5.78062 10.47L5.69625 10.3494C5.39682 9.91913 5.11533 9.47667 4.8525 9.02313C4.59045 8.56923 4.34733 8.10466 4.12375 7.63063L4.06125 7.4975L4.12375 7.36438C4.34729 6.89033 4.59042 6.42576 4.8525 5.97188C5.11125 5.52438 5.39563 5.07813 5.69625 4.64563L5.78062 4.525L5.9275 4.5125C6.97416 4.42375 8.02646 4.42375 9.07312 4.5125L9.21938 4.525L9.30313 4.64563C9.90261 5.50629 10.4288 6.41573 10.8763 7.36438L10.9394 7.4975L10.8763 7.63063C10.4299 8.57984 9.90363 9.48936 9.30313 10.3494L9.21938 10.47L9.07312 10.4825C8.54375 10.5263 8.01437 10.5488 7.5 10.5488ZM6.12688 9.87125C7.05188 9.94063 7.94812 9.94063 8.87375 9.87125C9.39052 9.11578 9.84933 8.32227 10.2462 7.4975C9.85003 6.67224 9.39097 5.87865 8.87313 5.12375C7.95902 5.05438 7.04098 5.05438 6.12688 5.12375C5.6088 5.87851 5.14973 6.67212 4.75375 7.4975C5.1511 8.32217 5.61011 9.11567 6.12688 9.87125Z" fill="#F1F1F1"/>
</svg>
