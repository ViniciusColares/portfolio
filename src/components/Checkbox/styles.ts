import styled from 'styled-components'
import css from '@styled-system/css'
import { motion, MotionProps } from 'framer-motion'

import { colors } from '@styles/theme'

export const CheckboxContainer = styled('label')(
  css({
    position: 'relative',
    display: 'flex',
    alignItems: 'center',
    userSelect: 'none',
    fontFamily: 'text',
    marginBottom: 0,
    cursor: 'pointer',
    span: {
      color: 'white',
      fontSize: 1,
      display: 'block',
      lineHeight: 1.3,
      ':first-letter': {
        textTransform: 'capitalize'
      }
    }
  })
)

export const CheckboxStyle = styled(motion.input)<MotionProps>(
  css({
    all: 'unset',
    position: 'relative',
    bg: 'primaryDark',
    minWidth: '18px',
    height: '18px',
    border: 'none',
    borderRadius: '4px',
    borderColor: 'primaryDark',
    display: 'inline-block',
    marginRight: '10px',
    transition: 'all 150ms',
    '&:focus, &:hover': {
      boxShadow: `inset 0px 0px 0px 2px ${colors.primaryLight}`
    },
    '&:checked': {
      boxShadow: `inset 0px 0px 0px 2px ${colors.aux.success}`,
      '::before': {
        content: '""',
        position: 'absolute',
        left: '6px',
        top: '6px',
        width: '6px',
        height: '6px',
        backgroundColor: 'aux.success',
        borderRadius: '1px'
      }
    },
    '&:disabled': {
      boxShadow: `inset 0px 0px 0px 2px ${colors.gray50}`,
      bg: 'gray25',
      cursor: 'not-allowed',
      '&:focus, &:hover': {
        boxShadow: `inset 0px 0px 0px 2px ${colors.primaryLight}`
      }
    }
  })
)
