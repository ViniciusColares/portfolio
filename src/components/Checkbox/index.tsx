import { useState, useEffect } from 'react'
import { MotionProps } from 'framer-motion'

import * as $ from './styles'

export const Checkbox = ({
  id,
  checked = false,
  onChange,
  group,
  disabled,
  label
}: CheckboxProps & React.HTMLAttributes<HTMLInputElement> & MotionProps) => {
  const [checkedBox, setCheckedBox] = useState(checked)

  const callAction = (e) => {
    setCheckedBox(!checkedBox)
    onChange && onChange(e)
  }

  useEffect(() => {
    setCheckedBox(checked)
  }, [checked])

  return (
    <$.CheckboxContainer htmlFor={`checkbox${id}`}>
      <$.CheckboxStyle
        id={`checkbox${id}`}
        onChange={(e) => !disabled && callAction(e)}
        name={group}
        value={id}
        checked={checkedBox}
        disabled={disabled}
        type="checkbox"
      />
      <span>{label}</span>
    </$.CheckboxContainer>
  )
}

export interface CheckboxProps {
  checked?: boolean
  group?: string
  disabled?: boolean
  label: string
}

export default Checkbox
