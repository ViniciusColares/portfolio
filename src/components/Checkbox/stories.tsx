import React, { ComponentProps } from 'react'
import { Story, Meta } from '@storybook/react/types-6-0'

import Checkbox from '.'

export default {
  title: 'Core/Checkbox',
  component: Checkbox
} as Meta

const Template: Story<ComponentProps<typeof Checkbox>> = (args) => (
  <Checkbox {...args} />
)

export const Basic = Template.bind({})
Basic.args = {
  name: 'Checkbox'
}
Basic.argTypes = { onClick: { action: 'clicked' } }

export const Loading = Template.bind({})
Loading.args = {
  name: 'Loading',
  isLoading: true
}
