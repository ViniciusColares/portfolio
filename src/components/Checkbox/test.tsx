import { screen, render } from '@testing-library/react'

import Checkbox from '.'

describe('Checkbox', () => {
  it('it should render a Checkbox with label', () => {
    const { container } = render(<Checkbox label="remember me" />)
    expect(screen.getByRole('checkbox'))
    expect(container.firstChild).toMatchSnapshot()
  })
})
