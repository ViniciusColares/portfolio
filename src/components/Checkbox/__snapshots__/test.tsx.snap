// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox it should render a Checkbox with label 1`] = `
.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  font-family: text;
  margin-bottom: 0;
  cursor: pointer;
}

.c0 span {
  color: white;
  font-size: 14px;
  display: block;
  line-height: 1.3;
}

.c0 span:first-letter {
  text-transform: capitalize;
}

.c1 {
  all: unset;
  position: relative;
  background-color: primaryDark;
  min-width: 18px;
  height: 18px;
  border: none;
  border-radius: 4px;
  border-color: primaryDark;
  display: inline-block;
  margin-right: 10px;
  -webkit-transition: all 150ms;
  transition: all 150ms;
}

.c1:focus,
.c1:hover {
  box-shadow: inset 0px 0px 0px 2px #B079FF;
}

.c1:checked {
  box-shadow: inset 0px 0px 0px 2px #00B007;
}

.c1:checked::before {
  content: "";
  position: absolute;
  left: 6px;
  top: 6px;
  width: 6px;
  height: 6px;
  background-color: success;
  border-radius: 1px;
}

.c1:disabled {
  box-shadow: inset 0px 0px 0px 2px #938CA3;
  background-color: gray25;
  cursor: not-allowed;
}

.c1:disabled:focus,
.c1:disabled:hover {
  box-shadow: inset 0px 0px 0px 2px #B079FF;
}

<label
  class="c0"
  for="checkboxundefined"
>
  <input
    class="c1"
    id="checkboxundefined"
    type="checkbox"
    value=""
  />
  <span>
    remember me
  </span>
</label>
`;
